pkg/server/image.go

```diff
@@ -337,19 +337,58 @@ func (ih *ImageHandler) scaleImage(ctx context.Context, fileRef blob.Ref) (*form
 var singleResize singleflight.Group
 
 func (ih *ImageHandler) ServeHTTP(rw http.ResponseWriter, req *http.Request, file blob.Ref) {
-	ctx := req.Context()
 	if !httputil.IsGet(req) {
 		http.Error(rw, "Invalid method", http.StatusBadRequest)
 		return
 	}
+
+	ctx := req.Context()
+	res, err := ih.Search.Describe(ctx, &search.DescribeRequest{
+		BlobRef: file,
+		Depth:   1,
+	})
```

Comment:
one downside of this change is that even for If-None-Match requests where we previously replied "304 Not Modified!" right away without doing any I/O, now we're doing a heavy Describe request just to do nothing with the result.